Src/Web/Web.Shared.Net/RequestTrackingTelemetryModule.cs

```diff
@@ -216,7 +240,11 @@ public void OnEndRequest(HttpContext context)
                 requestTelemetry.Source = telemetrySource;
             }
 
-            this.telemetryClient.TrackRequest(requestTelemetry);
+            string requestId = context.Request.Headers[RequestIdHeader];
+            if (requestId != null && activeRequests.TryRemove(requestId, out byte value))
```

Comment:
Why the failure to remove will lead to not tracked request? It's better to report two requests than zero in case you silently attempted to remove the header.

Another thing - add logging in case it failed to remove.